### PR TITLE
Create table builder structure

### PR DESCRIPTION
**Pull request by AugustoFKL**
Proposal of usage of a `CreateTableBuilder` structure, instead of the pure create table that we've been using right now.

This has 2 benefits:
- We can reduce calls anywhere the statement isn't meant to be "complete", i.e., when some parts are certain to be a default value.
- Callers can switch between the current statement to this structure, so they don't need to move an enum, a struct instead.

@alamb @nickolay what do you think? 

I know this resembles https://github.com/sqlparser-rs/sqlparser-rs/pull/315#issuecomment-1259597684 but, instead of replacing our current API, we can give the upstream the possibility of creating the struct, moving it with more ease around the code. If you think that's OK, I'll add a `TryFrom<Statement>` for this (or maybe just a convert method), and expand that going from the biggest statement to the smallest.

### CODE REVIEW

**Comment by alamb** [comment]
I think this makes a lot of sense to me 👍 

I think it needs some docstrings (ideally with an example) but otherwise 🚀

**Comment by AugustoFKL** [comment]
> I think this makes a lot of sense to me 👍
> 
> I think it needs some docstrings (ideally with an example) but otherwise 🚀

Ok! Will be done in 10 :)

**Comment by AugustoFKL** [comment]
@alamb done!

**Comment by alamb** [comment]
Looks like there is a minor issue with rustfmt/clippy

File: src/ast/helpers/stmt_create_table.rs [hunk 1]
```diff
@@ -0,0 +1,325 @@
...
+use crate::ast::{
+    ColumnDef, FileFormat, HiveDistributionStyle, HiveFormat, ObjectName, OnCommit, Query,
+    SqlOption, Statement, TableConstraint,
+};
+use crate::parser::ParserError;
+
+/// Builder for create table statement variant ([1]).
+///
+/// This structure is a facilitator to allow building and accessing a create table with more ease,
+/// without needing to:
```
Comment (alamb) [inline src/ast/helpers/stmt_create_table.rs:16]:
```suggestion
/// This structure helps building and accessing a `create table` with more ease,
/// without needing to:
```

File: src/ast/helpers/stmt_create_table.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+///        options: vec![],
+/// }]);
+/// // You can access internal elements with ease
+/// assert!(builder.if_not_exists);
+/// // Convert to a statement
+/// assert_eq!(
+///    builder.build().to_string(),
+///    "CREATE TABLE IF NOT EXISTS table_name (c1 INT)"
+/// )
+///
```
Comment (alamb) [inline src/ast/helpers/stmt_create_table.rs:39]:
```suggestion
```

File: src/ast/helpers/stmt_create_table.rs [hunk 3]
```diff
@@ -0,0 +1,325 @@
...
+    #[test]
+    pub fn test_from_valid_statement() {
+        let builder = CreateTableBuilder::new(ObjectName(vec![Ident::new("table_name")]));
+
+        let stmt = builder.clone().build();
+
+        assert_eq!(builder, CreateTableBuilder::try_from(stmt).unwrap());
+    }
+
+    #[test]
```
Comment (alamb) [inline src/ast/helpers/stmt_create_table.rs:314]: 👍

**Review by alamb — APPROVED** [review]
LGTM -- thanks @AugustoFKL

**Comment by alamb** [comment]
It just needs to get a clean CI and we can merge it.

**Pull request merged**